src/management-system-v2/app/(dashboard)/[environmentId]/layout-client.tsx

```diff
@@ -64,6 +75,36 @@ const Layout: FC<
   return (
     <UserSpacesContext.Provider value={userEnvironments}>
       <SpaceContext.Provider value={activeSpace}>
+        {userData && !userData.guest ? (
+          <AuthenticatedUserDataModal
+            modalOpen={!userData.username || !userData.lastName || !userData.firstName}
```

Comment:
Wouldn't it be more efficient to have these conditions added above to only render the modal when needed? I assume the modal is not gonna switch between open and close very often, where keeping it mounted would be beneficial.

Reply:
Yes, that is true, but there wouldn't be a closing animation if I did it that way. If you think component not being always mounted is more important, I can change that, if not, I should definitely set the `destroyOnClose` prop to true.

Reply:
Okay yeah, that is unfortunate. Definitely use that prop though, I agree.